de.metas.material/dispo-commons/src/main/java/de/metas/material/dispo/commons/interceptor/C_OrderLine.java

```diff
@@ -29,39 +18,26 @@
  *
  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
  * GNU General Public License for more details.
  *
  * You should have received a copy of the GNU General Public
- * License along with this program.  If not, see
+ * License along with this program. If not, see
  * <http://www.gnu.org/licenses/gpl-2.0.html>.
  * #L%
  */
 @Interceptor(I_C_OrderLine.class)
-@Component("MaterialDispoOrderLineInterceptor")
+@Component("de.metas.material.dispo.commons.interceptor.C_OrderLine")
 public class C_OrderLine
 {
-	@ModelChange(timings = {ModelValidator.TYPE_BEFORE_CHANGE, ModelValidator.TYPE_BEFORE_NEW })
-	public void updateQtyAvailableToPromise(final I_C_OrderLine ol)
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE }, ifColumnsChanged = { I_C_OrderLine.COLUMNNAME_M_Product_ID })
```

Comment:
hi @cristinamghita, if we update on M_Product_ID, let's also update on M_AttributeSetInstance_ID, and update all order lines if C_Order.PreparationDate changes